src/server/worker/api_server.go

```diff
@@ -288,6 +293,14 @@ func NewAPIServer(pachClient *client.APIClient, etcdClient *etcd.Client, etcdPre
 	if err != nil {
 		return nil, fmt.Errorf("error creating datum cache: %v", err)
 	}
+	exportStats := false
+	resp, err := pachClient.Enterprise.GetState(context.Background(), &enterprise.GetStateRequest{})
+	if err != nil {
+		// No logger yet, we need a server first
```

Comment:
I think it might make sense to move this below the line where we get the logger so we can log this in a way that can be consumed by `get-logs`, users are unlikely to see the message as it is now because they'll need to do `kubectl logs` rather than `get-logs` which most users don't do.

It also has the benefit that it makes the code a bit shorter, because you don't need an `exportStats` variable, you can just do `server.exportStats = resp.State == enterprise.State_ACTIVE`

Reply:
Ah ok. Yea I can do that. As a general practice I like to keep values that are used to initialize a struct all in one place when we create the new struct literal. Idk ... that just seems cleaner since that keeps the state defined all in one place. And so I didn't want to modify server after the fact. 

But, obviously `exportStats` is not something that really has any side effects other than the ones I want ... so nothing really to worry about in this case.

Reply:
It's always preferable to initialize everything in the struct in one place, particularly if you're declaring the struct inline. In cases like this one, where you're defining a constructor, I think the constructor is the one place that all the state is being kept in so the rule applies a little less. You know that no one is going to be touching the struct or experiencing sideeffects from it, since it's not accessible from outside the constructor.